### PR TITLE
[4.0] Drop for loops

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
Fix various problems on:
- plg_joomlaupdate
- plg_extensionupdate
- plg_overridecheck

### Testing Instructions
Apply patch and check if the three button work correctly
screenshot 2018-12-28 at 12 11 18

### Actual result
Problems: 
- Usage of for loops
- No check before calling a fn on a document element (for whatever reason that element might not exist)
- ESLint

### Documentation Changes Required
No

### CODE REVIEW

File: build/media_src/plg_quickicon_extensionupdate/js/extensionupdatecheck.es6.js [hunk 1]
```diff
@@ -11,27 +11,41 @@
...
-              }
+              // No updates
+              update('success', Joomla.Text._('PLG_QUICKICON_EXTENSIONUPDATE_UPTODATE'));
             } else {
               const messages = {
                 warning: [
-                  `<div class="message-alert">` + `${Joomla.JText._('PLG_QUICKICON_EXTENSIONUPDATE_UPDATEFOUND_MESSAGE').replace('%s', `<span class="badge badge-pill badge-danger">${updateInfoList.length}</span>`)}<button class="btn btn-sm btn-primary" onclick="document.location='${options.url}'">${Joomla.JText._('PLG_QUICKICON_EXTENSIONUPDATE_UPDATEFOUND_BUTTON')}</button>`,
+                  `<div class="message-alert">
+  ${Joomla.Text._('PLG_QUICKICON_EXTENSIONUPDATE_UPDATEFOUND_MESSAGE').replace('%s', `<span class="badge badge-pill badge-danger">${updateInfoList.length}</span>`)}
+  <button class="btn btn-sm btn-primary" onclick="document.location=${options.url}">
```
Comment (brianteeman) [inline build/media_src/plg_quickicon_extensionupdate/js/extensionupdatecheck.es6.js]: missing type=

Comment (dgrammatiko) [reply build/media_src/plg_quickicon_extensionupdate/js/extensionupdatecheck.es6.js]: fair enough

Comment (dgrammatiko) [reply build/media_src/plg_quickicon_extensionupdate/js/extensionupdatecheck.es6.js]: There is a closing div also missing here, blame whoever patched these files

**Comment by Fedik** [comment]
> Usage of for loops

it not a problem, but I don't  mind :smile:

**Comment by dgrammatiko** [comment]
> it not a problem

Actually there is: https://www.youtube.com/watch?v=Nzokr6Boeaw&list=PLNYkxOF6rcIAKIQFsNbV0JDws_G_bnNo9&index=12

**Comment by Fedik** [comment]
complicated != problem :wink:

**Comment by dgrammatiko** [comment]
True but also for() is slower in new browsers and not really compatible with Joomla's CS: https://github.com/airbnb/javascript

**Comment by Fedik** [comment]
> but also for() is slower in new browsers

nope, it is faster than any loop :wink: 
http://jsben.ch/gROW8
https://jsperf.com/for-vs-foreach/75

> and not really compatible with Joomla's CS

okay got it

**Comment by dgrammatiko** [comment]
> nope, it is faster than any loop 

New browsers also means ES6+ code, not vars...

**Comment by Fedik** [comment]
> New browsers also means ES6+ code, not vars...

nope, it still faster https://jsperf.com/for-vs-foreach-vs-for-of/575 :wink:

**Comment by wilsonge** [comment]
I think the readability trumps the small performance enhancement here. Although if we decide it practically doesn't work on scaling sites we can always come back to this :) This fixes various pieces of broken html so doing this for now

**Comment by infograf768** [comment]
See https://github.com/joomla/joomla-cms/pull/23434 where js is corrected.

**Pull request merged**